forest/src/cli/chain_cmd.rs

```diff
@@ -46,50 +46,36 @@ impl ChainCommands {
         match self {
             Self::Block { cid } => {
                 let cid: Cid = cid.parse().unwrap();
-                let mut client = new_client();
-
-                let blk = block(&mut client, cid)
-                    .await
-                    .map_err(stringify_rpc_err)
-                    .unwrap();
+                let blk = block(cid).await.map_err(stringify_rpc_err).unwrap();
                 println!("{}", serde_json::to_string_pretty(&blk).unwrap());
             }
             Self::Genesis => {
-                let mut client = new_client();
-
-                let gen = genesis(&mut client)
-                    .await
-                    .map_err(stringify_rpc_err)
-                    .unwrap();
+                let gen = genesis().await.map_err(stringify_rpc_err).unwrap();
                 println!("{}", serde_json::to_string_pretty(&gen).unwrap());
             }
             Self::Head => {
-                let mut client = new_client();
-
-                let canonical = head(&mut client).await.map_err(stringify_rpc_err).unwrap();
+                let canonical = head().await.map_err(stringify_rpc_err).unwrap();
                 println!(
                     "{}",
-                    serde_json::to_string_pretty(&canonical.0.cids()).unwrap()
+                    serde_json::to_string_pretty(
+                        &canonical
+                            .0
+                            .cids()
+                            .iter()
+                            .map(|cid: &Cid| cid.to_string())
+                            .collect::<Vec<String>>()
```

Comment:
In case you find it interesting: often times when running collect, the type inference engine just needs to know that you want to use the `Vec<_>` implementation, and can easily detect the `T` within the `Vec<_>`